### PR TITLE
[opt] initialize HLSL passes with SetupRegistryPassForHLSL

### PR DESCRIPTION
**Pull request by python3kgae**
call hlsl::SetupRegistryPassForHLSL in opt to initialize all HLSL/DXIL passes.

This is for FileCheck test which use opt to test HLSL/DXIL passes.

### CODE REVIEW

File: tools/opt/opt.cpp [hunk 1]
```diff
@@ -361,10 +364,8 @@ int __cdecl main(int argc, char **argv) {
   //initializeDwarfEHPreparePass(Registry); // HLSL Change: remove EH passes
   //initializeSjLjEHPreparePass(Registry);  // HLSL Change: remove EH passes
   // HLSL Change Starts
-  initializeReducibilityAnalysisPass(Registry);
```
Comment (pow2clk) [inline tools/opt/opt.cpp:364]: This pass is actually initialized a few lines above as well. It was redundant then and redundant now, but we might as well remove it.

**Review by tex3d — APPROVED** [review]

File: tools/opt/CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,8 @@ set(LLVM_LINK_COMPONENTS
   DXIL # HLSL Change
   DxcBindingTable # HLSL Change
   HLSL # HLSL Change
+  DxilContainer #  HLSL Change for DxcOptimizerPass
+  DxilRootSignature #  HLSL Change for DxcOptimizerPass
```
Comment (pow2clk) [inline tools/opt/CMakeLists.txt:11]: I didn't look into exactly which passes require these, but I take your word for it 😄

**Review by pow2clk — APPROVED** [review]
I would prefer to remove the redundant pass, but it's no worse than before. It's definitely right to initialize all the passes instead of piecemeal as they are needed

File: tools/opt/opt.cpp (hunk 1 above)
Comment (python3kgae) [reply tools/opt/opt.cpp:364]: Good catch.
Fixed.

File: tools/opt/CMakeLists.txt (hunk 2 above)
Comment (python3kgae) [reply tools/opt/CMakeLists.txt:11]: I'm not sure which pass too. But I'm sure without these two libs the linking will fail.

**Pull request merged**